File: src/python/transaction/operations/account_update/change_weight_threshold.py
import asyncio

from wax import create_hive_chain
from wax.complex_operations.account_update import AccountAuthorityUpdateOperation


async def main() -> None:
    # Initialize hive chain interface
    wax = create_hive_chain()

    # Initialize a transaction object
    tx = await wax.create_transaction()

    # Create online operation - createFor will parse authorities for "gtg" account from the chain
    op = await AccountAuthorityUpdateOperation.create_for(wax, account_name="gtg")

    # Add "initminer" to owner role and change weight threshold to 2
    op.roles.owner.add("initminer", 1).set_threshold(threshold=2)

    # Push operation to transaction
    tx.push_operation(op)
    print(f"Transaction ready to sign and broadcast: {tx.to_string()}")


asyncio.run(main())

File: src/rust/transaction/operations/account_update/change_weight_threshold.rs
use wax::prelude::*;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // Initialize hive chain interface
    let chain = create_hive_chain(None)?;

    // Initialize a transaction object
    let mut tx = chain.create_transaction(None).await?;

    // Create online operation - create_for will parse authorities for "gtg" account from the chain
    let mut op = AccountAuthorityUpdateOperation::create_for(&chain, "gtg").await?;

    // Add "initminer" to owner role and change weight threshold to 2
    op.owner.add("initminer", 1)?.set_threshold(2)?;

    // Push operation to transaction
    tx.push_complex_operation(&chain, op)?;

    Ok(())
}
